ros2topic/ros2topic/verb/pub.py

```diff
@@ -64,13 +64,23 @@ def publisher(message_type, topic_name, values):
     pub = node.create_publisher(msg_module, topic_name)
 
     msg = msg_module()
-    for field_name, field_value in values_dictionary.items():
-        field_type = type(getattr(msg, field_name))
-        setattr(msg, field_name, field_type(field_value))
+    set_msg_fields(msg, values_dictionary)
 
     print('publisher: beginning loop')
     while rclpy.ok():
         pub.publish(msg)
         print('publishing %r\n' % msg)
         time.sleep(1)
     rclpy.shutdown()
+
+
+def set_msg_fields(msg, values):
+    for field_name, field_value in values.items():
+        print(field_name, field_value)
+        field_type = type(getattr(msg, field_name))
+        try:
+            value = field_type(field_value)
+        except TypeError:
+            value = field_type()
+            set_msg_fields(value, field_value)
+        setattr(msg, field_name, value)
```

Comment:
If the assert would contain a reasonable message (which I added for this manually) I get the following now:

> Failed to populate field 'covariance': The 'covariance' field must be a seq or list with length 36 and float values